subscription_manager/utils.py

```diff
@@ -1,7 +1,56 @@
-from typing import List
+from dataclasses import fields
+from typing import Tuple
+from random import choice, uniform, randint
+from datetime import datetime, date
+from subscription_manager.exceptions import *
+from subscription_manager.subscription import Subscription
 
 
-def _validate_subscription(subscription: dict) -> bool:
+def _get_currencies_list() -> Tuple[str]:
+    """
+    Return list of supported currencies.
+    Returns:
+        tuple: list of supported currencies
+    """
+    return 'USD', 'GBP', 'EUR', 'RUB', 'CNY'
+
+
+def _get_frequencies_list() -> Tuple[str]:
+    """
+    Return list of supported frequencies.
+    Returns:
+        tuple: list of supported currencies
+    """
+    return 'daily', 'weekly', 'monthly', 'yearly'
+
+
+def subscription_generator() -> dict:
+    """
+    Generate subscription with correct fields.
+    Returns:
+        subscription (dict): generated subscription
+    """
+    names = ['Mary', 'Eva', 'Nancy', 'Ann', 'Kevin', 'John', 'Ben', 'Robin']
+    currencies = _get_currencies_list()
+    frequencies = _get_frequencies_list()
+    subscription_names = ['Google music', 'Spotify', 'Apple Music', 'Youtube Music', 'Coursera', 'Pluralsight',
+                          'Amazon Prime', 'Netflix', 'Sky Store', 'Pet insurance', 'Mobile payment']
+    # generate random day from the beginning of the current year to today
+    start_dt = date.today().replace(day=1, month=1).toordinal()
+    end_dt = date.today().toordinal()
+    random_day = date.fromordinal(randint(start_dt, end_dt))
+
+    return dict(owner=choice(names),
+                name=choice(subscription_names),
+                frequency=choice(frequencies),
+                start_date=random_day,
+                price=round(uniform(1.99, 49.99), 2),
+                currency=choice(currencies),
+                comment="Generation date: " + datetime.today().strftime("%d/%m/%Y, %H:%M:%S")
+                )
+
+
+def validate_subscription(subscription: dict) -> bool:
```

Comment:
Can we please replace it with create_subscription(**fields):

I expected it to raise exception in case of incorrect parameters

Reply:
done